lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx

```diff
@@ -0,0 +1,219 @@
+                                                <DeleteIcon />
+                                            </button>
+                                        </div>
+
+                                        <MetadataEditModal index={index} metadata={metadata} modal={modal} />
+                                    </td>
+                                </tr>
+                            );
+                        })}
+                    </tbody>
+                </table>
+                <div className={'flex justify-end'}>
+                    <button
+                        className='btn'
+                        onClick={() => {
+                            addNewMetadata();
+                        }}
+                    >
+                        <AddIcon />
+                    </button>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+export function MetadataEditModal({ index, metadata, modal }: { index: number; metadata: Metadata; modal: string }) {
+    const { updateMetadata } = useContext(ConfigContext);
+
+    const [metadataType, setMetadataType] = useState<MetadataType>(metadata.type);
+    const [metadataName, setMetadataName] = useState(metadata.name);
+    const [generateIndex, setGenerateIndex] = useState(metadata.generateIndex);
+
+    const handleUpdateMetadata = () => {
+        updateMetadata({ ...metadata, name: metadataName, type: metadataType, generateIndex }, index);
+    };
+
+    return (
+        <dialog id={modal} className='modal'>
+            <div className='modal-box'>
+                <h3 className='font-bold text-lg'>Edit metadata</h3>
+                <div className='flex flex-col space-x-2 items-center'>
+                    <table className='table'>
+                        <tbody>
+                            <tr>
+                                <td>Name</td>
+                                <td>
+                                    <EditMetadataName metadataName={metadataName} setMetadataName={setMetadataName} />
+                                </td>
+                            </tr>
+
+                            <tr>
+                                <td>Type</td>
+                                <td>
+                                    <SelectMetadataType metadataType={metadataType} setMetadataType={setMetadataType} />
+                                </td>
+                            </tr>
+
+                            <tr>
+                                <td>Generate Index</td>
+                                <td>
+                                    <EditGenerateIndex
+                                        generateIndex={generateIndex}
+                                        setGenerateIndex={setGenerateIndex}
+                                    />
+                                </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                </div>
+                <form method='dialog'>
```

Comment:
If we put the input fields inside this form, we might get confirming by pressing Enter for free.

Reply:
This does only work for the text field, but not the others edits. 
I would also argue, that the extra click to confirm does prevent the user to press enter after typing (when trying to get the focus away), and then submitting the data.

Reply:
As discussed: we add a new ticket for this. #618